execution/executor-benchmark/src/transaction_generator.rs

```diff
@@ -405,7 +411,7 @@ impl TransactionGenerator {
     }
 
     /// Generates transactions for random pairs of accounts.
-    pub fn gen_transfer_transactions(
```

Comment:
@manudhundi - Sorry about the confusion but This is old way of generating txns, which should be deprecated. Can you add this functionality to the  transaction generator in `run_workload` function ?

Reply:
May be we can add a new sampling mode here https://github.com/aptos-labs/aptos-core/blob/4cc1c40a1b5b27c71b79a976458b27f3a7629541/crates/transaction-generator-lib/src/p2p_transaction_generator.rs#L25. @zjma thoughts?

Reply:
IMO we should stop messing with `TransactionGenerator`s for controlling in-block conflict, because [its API](https://github.com/aptos-labs/aptos-core/blob/991c7ed1a8375f8f799583b628d5a8974656ef35/crates/transaction-generator-lib/src/lib.rs#L85-L91) is not designed for this: e.g. non-conflicting block generating was trivial, but we made it much more complicated by `BurnAndRecycle` sampling inside TransactionGenerator, and outside also ensuring unique senders as well as correct pool size... (remember why pool size has to be 3x+ of block size?)
 
We simply need a trait `BlockGenerator` with API `fn generator_block(size: usize) -> Vec<Transaction>`.
And we can have instances
```
block_generators::P2PDefault {
  accounts: Vec<Account>,
}

block_generators::P2PAllConnected { // connected transfer graph
  accounts: Vec<Account>,
}

///  Can leverage P2PAllConnected;
/// get non-conflicting block for free when num_non_conflicting_groups=block_size
block_generators::P2PNonConflictGroup {
  accounts: Vec<Account>,
  num_non_conflicting_groups: usize,
}

block_generators::P2PHotspot {
  accounts: Vec<Account>,
  hotspot: f32, //if 0.8, then 80% accounts will be picked with probability 20%, the other 20% get 80%
}

// other patterns/shapes
```

Reply:
As discussed offline, adding the functionality in `TransactionGenerator` used by `run_workload` requires non-trivial amount of refactoring. So adding it here (in `run_transfer`) for now. We will get a refactoring in the future perhaps

@zjma trait `BlockGenerator` is a good idea, considered this as part of the above [refactoring](https://github.com/aptos-labs/aptos-core/commit/719953622c9d0236ed427cb26edc7b892c88318c). But, to `run_workload` with `BlockGenerator` we will have to refactor `TransactionGenerator` and also add `BlockGenerator` for non P2P cases.